pkg/docker/parseimage.go

```diff
@@ -24,25 +24,61 @@ package docker
 import (
 	"fmt"
 	"regexp"
+	"strings"
 )
 
-var prefix = regexp.MustCompile("docker-pullable://")
-var digestRegexp = regexp.MustCompile("@sha256:([a-zA-Z0-9]+)$")
+var dockerPullableRegexp = regexp.MustCompile("^docker-pullable://(.+)@sha256:([a-zA-Z0-9]+)$")
+var dockerRegexp = regexp.MustCompile("^docker://sha256:([a-zA-Z0-9]+)$")
+var imageRegexp = regexp.MustCompile("^(.+)@sha256:([a-zA-Z0-9]+)$")
 
 // ParseImageIDString parses an ImageID that can pull an image from docker
 // Example image id:
 //   docker-pullable://registry.kipp.blackducksoftware.com/blackducksoftware/hub-registration@sha256:cb4983d8399a59bb5ee6e68b6177d878966a8fe41abe18a45c3b1d8809f1d043
 func ParseImageIDString(imageID string) (string, string, error) {
-	str := imageID
-	match := prefix.FindStringIndex(str)
-	if len(match) > 0 && match[0] == 0 {
-		str = str[match[1]:]
+	name, digest, err := ParseDockerPullableImageString(imageID)
+	// Since the GO doesn't support for "don't start with string regex", had an ugly fix with HasPrefix in below line
+	if err == nil || strings.HasPrefix(imageID, "docker-pullable://") {
+		return name, digest, err
 	}
-	match2 := digestRegexp.FindStringSubmatchIndex(str)
-	if len(match2) != 4 {
-		return "", "", fmt.Errorf("unable to match digestRegexp regex <%s> to input <%s>", digestRegexp.String(), str)
+
+	name, digest, err = ParseImageString(imageID)
+	if err == nil {
+		return name, digest, err
+	}
+
+	name, digest, err = ParseDockerImageString(imageID)
+	if err == nil {
+		return name, digest, fmt.Errorf("Scanning of static/builder pod image %s is not supported!!", imageID)
```

Comment:
Static pods is only one case where this might occur.  It might be the only one, but I'm not sure.  Better not to call out the specific failure case by name

Reply:
Removed static/builder from error message